packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -710,24 +769,79 @@ const runGenerator = (context, {outputDir, templateVersion, verbose}) => {
         const assetsDir = p.join(ASSETS_TEMPLATES_DIR, id)
         if (sh.test('-e', assetsDir)) {
             getFiles(assetsDir)
-                .map((file) => file.replace(assetsDir, ''))
-                .forEach((relFilePath) =>
+                .map((file) => {
+                    const relFilePath = file.replace(assetsDir, '')
+                    return relFilePath
+                })
+                .forEach((relFilePath) => {
                     processTemplate(relFilePath, assetsDir, outputDir, context)
+                })
+        }
+
+        let extensionSamplePath
+        if (extractExtension) {
+            const extensionTmp = fs.mkdtempSync(p.resolve(os.tmpdir(), 'extract-extension-sample'))
+            const extensionTarFile = sh
+                .exec(
+                    `npm pack @salesforce/extension-sample@1.0.0-dev --pack-destination="${extensionTmp}"`,
+                    {
+                        silent: true
+                    }
                 )
+                .stdout.trim()
+            const extensionTarPath = p.join(extensionTmp, extensionTarFile)
+
+            // Extract the extension
+            const extensionFolder = p.join(outputDir, 'extension-sample')
+            tar.x({
+                file: extensionTarPath,
+                cwd: extensionTmp,
+                sync: true
+            })
+
+            // Copy the extension-sample into the output directory
+            extensionSamplePath = p.join(extensionTmp, 'package')
+            sh.cp('-rf', extensionSamplePath, outputDir)
+            sh.mv(extensionSamplePath, extensionDir)
+
+            // Clean up extension temporary directory
+            sh.rm('-rf', extensionTmp)
+
+            // Set the relative path for the extracted extension
```

Comment:
This code might be a good candidate for a util of some kind, or maybe slap it in it's own function so it's better documentable and in the future we can generalize it.